configure

```diff
@@ -0,0 +1,385 @@
+  TEST_CXXFLAGS                 If you want to use some arguments in C++ compiler for testing.
+  TEST_CXXFLAGS_EXTRA           If you want to use some arguments in C++ compiler for testing.
+                                  It's passed after TEST_CXXFLAGS without overwrite it.
+  CPPFLAGS                      If you want to use some arguments in C++ compiler preprocessor.
+  LDFLAGS                       If you want to use some arguments in Linker command line.
+  LDFLAGS_EXTRA                 If you want to use some arguments in Linker command line.
+                                  It's passed after LDFLAGS without overwrite it.
+  TEST_LDFLAGS                  If you want to use some arguments in Linker for testing.
+  TEST_LDFLAGS_EXTRA            If you want to use some arguments in Linker for testing.
+                                  It's passed after TEST_LDFLAGS without overwrite it.
+
+Note: be careful about passing a path with space as prefix/eprefix/sbindir/mandir:
+for this thing You need to use backslash (\) with double quotation (") for every space.
+For example: ./configure --sbindir="/usr/bin\ with\ space/"
+
+See BUILD file for more information.
+EOF
+    exit 0
+}
+
+## Flag parser
+for arg in "$@"; do
+    case "$arg" in
+        --help) usage ;;
+        --quiet)
+            info() { true; }
+            sub_info() { true; }
+            warning() { true; }
+            sub_warning() { true; }
+        ;;
+        --clean) rm -f test* & rm -f mconfig && exit 0 ;;
+        --platform=*) PLATFORM="${arg#*=}" && shift ;;
+        --prefix=*) PREFIX="${arg#*=}" && shift ;;
+        --exec-prefix=*) EPREFIX="${arg#*=}" && shift;;
+        --sbindir=*) SBINDIR="${arg#*=}" && shift ;;
+        --mandir=*) MANDIR="${arg#*=}" && shift ;;
+        --syscontrolsocket=*) SYSCONTROLSOCKET="${arg#*=}" && shift ;;
+        --shutdown-prefix=*) SHUTDOWN_PREFIX="${arg#*=}" && shift ;;
+        --enable-shutdown|--enable-shutdown=yes) BUILD_SHUTDOWN=yes ;;
+        --disable-shutdown|--enable-shutdown=no) BUILD_SHUTDOWN=no ;;
+        --enable-cgroups|--enable-cgroups=yes) SUPPORT_CGROUPS=1 ;;
+        --disable-cgroups|--enable-cgroups=no) SUPPORT_CGROUPS=0 ;;
+        --enable-sanitize|--enable-sanitize=yes) NO_SANITIZE=0 ;;
+        --disable-sanitize|--enable-sanitize=no) NO_SANITIZE=1 ;;
+        --enable-utmpx|--enable-utmpx=yes) USE_UTMPX=1 ;;
+        --disable-utmpx|--enable-utmpx=no) USE_UTMPX=0 ;;
+        CXX=*|CXX_FOR_BUILD=*|CXXFLAGS_FOR_BUILD=*|CPPFLAGS_FOR_BUILD=*\
+        |LDFLAGS_FOR_BUILD=*|CXXFLAGS=*|CXXFLAGS_EXTRA=*|TEST_CXXFLAGS=*\
+        |TEST_CXXFLAGS_EXTRA|LDFLAGS=*|LDFLAGS_EXTRA=*|TEST_LDFLAGS=*\
+        |TEST_LDFLAGS_EXTRA=*|CPPFLAGS=*)
+            envvar=$(expr "x$arg" : 'x\([^=]*\)=')
+            envvar_value=$(expr "X$arg" : '[^=]*=\(.*\)') || envvar_value="" # Fix empty var bug
+            eval "$envvar=\$envvar_value"
+            export "${envvar?}"
```

Comment:
Now you have restored the `export` as well. Why? It's not needed.

Reply:
When I just use `eval`, Users can't pass multi flags as `CXXFLAGS` & ....
I think this `export` is a simple way to fix this thing.

Reply:
Can you give a full example that doesn't work without the `export`? What I've tried works for me without `export`.

(`export` exports the environment variables to subprocesses, I can't see why that would make any difference here.).

Before, when you removed `export`, you also removed the `\` in the `eval` line, which is why it stopped working then. Are you getting mixed up due to that perhaps? You need to remove `export` but keep the `eval` line exactly as it is.

Reply:
I think that's `\` for be processed by `export` but seems like I did wrong.